client/src/components/Account/EditProfileForm/EditProfileForm.tsx

```diff
@@ -1,52 +1,140 @@
+    firstName: loggedInUser?.profile.firstName,
+    lastName: loggedInUser?.profile.lastName,
+    email: loggedInUser?.email,
+    phoneNumber: loggedInUser?.profile.phoneNumber,
+    address: loggedInUser?.profile.address,
+    description: loggedInUser?.profile.description,
+  });
+
+  const [showPhoneInput, setShowPhoneInput] = useState(profile?.phoneNumber ? true : false);
+  const handleChange = (
+    e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>,
+    property: string,
+  ): void => {
+    setProfile({ ...profile, [property]: e.target.value });
+  };
+
+  const toggleInput = () => setShowPhoneInput(!showPhoneInput);
+
+  const handleSaveProfile = async () => {
+    const id = loggedInUser ? loggedInUser?.profile._id : '';
+    try {
+      await updateProfile(id, profile);
+      updateSnackBarMessage('Profie updated');
+    } catch (error) {
+      updateSnackBarMessage(`Error updating user profile ${error}`);
+    }
+  };
+
   return (
     <Box>
       <Typography variant="h4" align="center" className={classes.formTitle}>
         Edit Profile
       </Typography>
       <form>
         <Grid container spacing={3}>
-          <CustomTextField onChange={handleChange} value="" label="FIRST NAME" placeholder="First Name" />
-          <CustomTextField onChange={handleChange} value="" label="LAST NAME" placeholder="Last Name" />
-          <CustomTextField onChange={handleChange} value="" label="EMAIL ADDRESS" placeholder="user@gmail.com" />
+          <CustomTextField
+            onChange={(e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) =>
+              handleChange(e, 'firstName')
+            }
+            value={profile.firstName ? profile.firstName : ''}
+            label="FIRST NAME"
+            placeholder="First Name"
+          />
+          <CustomTextField
+            onChange={(e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) =>
+              handleChange(e, 'lastName')
+            }
+            value={profile.lastName ? profile.lastName : ''}
+            label="LAST NAME"
+            placeholder="Last Name"
+          />
+          <CustomTextField
+            onChange={(e: React.ChangeEvent<HTMLInputElement> | React.ChangeEvent<HTMLTextAreaElement>) =>
+              handleChange(e, 'email')
+            }
+            value={profile.email ? profile.email : ''}
+            label="EMAIL ADDRESS"
+            placeholder="user@gmail.com"
+          />
           <Grid item xs={12}>
             <Grid container alignItems="center" spacing={2}>
               <Grid item xs={12} sm={3}>
                 <Typography variant="body1" align="right" className={classes.formLabel}>
                   PHONE NUMBER
                 </Typography>
               </Grid>
-              <Grid item xs={12} sm={4}>
+              <Grid item xs={12} sm={4} className={`${showPhoneInput && classes.shouldNotDisplay}`}>
```

Comment:
className shows a warning when it's equal to null/undefined, might want to change this to a ternary where the else case is `" "`.

Ditto for below